
# A Preliminary Investigation into the Evolution of a Near-Chaotic Genome

This project utilizes the wealth of genomic data generated by the [LTEE study](https://the-ltee.org/). The NCBI page for the umbrella project is found [here](https://www.ncbi.nlm.nih.gov/bioproject/294072).

The script order_index.py is an implementation of the order index algorithm as described in [*Genomes: at the edge of chaos with maximum information capacity*](https://arxiv.org/pdf/0708.1598) by Kong et al. The Python script is used to determine the level of orderliness of a genome, and this project was built to explore changes in the orderliness over the thousands of generations of *E. Coli* bacteria grown and maintained by the LTEE study.

## In order to reproduce the study:

[Python3](https://www.python.org/downloads/) is required to run the main script.

For the required data preprocessing, you will need to install [breseq](https://github.com/barricklab/breseq) ([bowtie2](https://bowtie-bio.sourceforge.net/bowtie2/index.shtml) and [R](https://www.r-project.org/) are required dependencies). The commands included in this file assume the breseq and gdtools binaries are located in ~/breseq/bin. If breseq is installed in a different location, be sure to adjust the path accordingly.

Download the [REL606 GenBank](https://www.ncbi.nlm.nih.gov/nuccore/NC_012967.1?report=gbwithparts&log$=seqview) file. This is the original strain that serves as the reference to compare subsequent generations to. Copy the genomic sequence out of the GenBank file and save it into a .txt file. This will be used for order_index.py.

Use NCBI's [Run Browser](https://trace.ncbi.nlm.nih.gov/Traces/?view=run_browser&display=metadata) to search for the SRA Run Accessions listed below. Download the fastq from each SRA's FASTA/FASTQ Download tab.

Next, follow these steps to generate the plain text genome sequence file needed for the order_index.py algorithm:

`./breseq/bin/breseq -r path/to/REL606.gbk -j 8 path/to/downloaded/fastq/XXXXXXXXX-YYK.fastq.gz -o path/to/desired/output/location/XXXXXXXXX-YYK`

Replace the XXXXXXXXX with the SRR ID, and the YY with the generation number (which is in thousands). Example: SRR098027-33k. The -j option indicates the number of processing cores the command is allowed to use.

This can take a while and range from several minutes to nearly an hour. There is a signficant amount of data and files generated with each run.

The output file from the previous process is then incorporated into this command, which generates the fasta file:

`./breseq/bin/gdtools APPLY -r path/to/REL606.gbk path/to/desired/output/location/XXXXXXXXX-YYK/output/output.gd -o path/to/final/output/location/XXXXXXXXX-YYK/gdtools_output_XXXXXXXXX-YYK.fasta`

Finally, duplicate the fasta file, remove the accession number from the first line, and change the file extension to .txt. Now we have a .txt file whose only contents are the genome nucleotide sequence, as required by order_index.py.

Now simply change the file path on line 9 of order_index.py as needed and run the script for the REL606 .txt file and subsequent generation .txt files. Analyze as desired.

### The following is the list of SRRs by strain and generation:

**ara-3**
- SRR098285 (20k)
- SRR098286 (20k)
- SRR098289 (31.5k)
- SRR098032 (32k)
- SRR098042 (32k)
- SRR098043 (32k)
- SRR098027 (33k)
- SRR098034 (34k)
- SRR098035 (34k)
- SRR098036 (36k)
- SRR098037 (36k)
- SRR098038 (38k)
- SRR098039 (38k)
- SRR098029 (40k)
- SRR098030 (40k)

**ara-6**
- SRR12340686 (50k)
- SRR12340691 (50k)
- SRR12340695 (50k)
- SRR12340704 (50k)

**ara+4**
- SRR12340689 (50k)
- SRR12340694 (50k)
- SRR12340698 (50k)
- SRR12340707 (50k)






